### PR TITLE
Clean app

### PR DESCRIPTION
**Pull request by sgliner-ledger**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,188 +15,92 @@
...
-DEFINES   += UI_NANO_S
+# Enable StageNet by default
+#DEFINES += MONERO_ALPHA // This will also disable mainnet
+#DEFINES += MONERO_BETA
+
+
+ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOS TARGET_NANOS2))
+DEFINES   += UI_NANO_SX
+else ifeq ($(TARGET_NAME),TARGET_NANOX)
+DEFINES   += UI_NANO_X
```
[the remark this answers is not in the document]
Comment (xchapron-ledger) [reply Makefile:74]: probably this defines could be dropped

File: src/monero_dispatch.c [hunk 2]
```diff
@@ -115,13 +100,13 @@ int check_ins_access() {
...
-            if (os_global_pin_is_validated() != PIN_VERIFIED) {
-                return SW_SECURITY_PIN_LOCKED;
-            }
             if (G_monero_vstate.tx_in_progress != 1) {
                 return SW_COMMAND_NOT_ALLOWED;
             }
             return SW_OK;
+        case INS_LOCK_DISPLAY:
+            // Deprecated command
+            return SW_INS_NOT_SUPPORTED;
```
Comment (xchapron-ledger) [inline src/monero_dispatch.c:109]: Do you known if many software wallets use this?

File: src/monero_init.c [hunk 3]
```diff
@@ -230,16 +266,8 @@ int monero_apdu_reset() {
...
 /* ----------------------------------------------------------------------- */
-int monero_apdu_lock() {
-    monero_io_discard(0);
-    monero_lock_and_throw(SW_SECURITY_LOCKED);
-    return SW_SECURITY_LOCKED;
-}
-
-void monero_lock_and_throw(int sw) {
-    G_monero_vstate.protocol_barrier = PROTOCOL_LOCKED;
+void monero_lock(int sw) {
```
Comment (xchapron-ledger) [inline src/monero_init.c:269]: This function is unused

File: src/monero_io.c [hunk 4]
```diff
@@ -144,25 +144,25 @@ void monero_io_insert_encrypt(unsigned char* buffer, int len, int type) {
 
 void monero_io_insert_u32(unsigned int v32) {
     monero_io_hole(4);
-    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 0] = v32 >> 24;
-    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 1] = v32 >> 16;
-    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 2] = v32 >> 8;
-    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 3] = v32 >> 0;
+    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 0] = (unsigned char)(v32 >> 24) & 0xFF;
```
Comment (xchapron-ledger) [inline src/monero_io.c:147]: Could be done with U4BE_ENCODE from SDK

File: src/monero_io.c [hunk 5]
```diff
@@ -144,25 +144,25 @@ void monero_io_insert_encrypt(unsigned char* buffer, int len, int type) {
...
+    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 1] = (unsigned char)(v32 >> 16) & 0xFF;
+    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 2] = (unsigned char)(v32 >> 8) & 0xFF;
+    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 3] = (unsigned char)(v32 >> 0) & 0xFF;
     G_monero_vstate.io_offset += 4;
 }
 
 void monero_io_insert_u24(unsigned int v24) {
```
Comment (xchapron-ledger) [inline src/monero_io.c:154]: never used

File: src/monero_io.c [hunk 6]
```diff
@@ -144,25 +144,25 @@ void monero_io_insert_encrypt(unsigned char* buffer, int len, int type) {
...
+    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 1] = (unsigned char)(v24 >> 8) & 0xFF;
+    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 2] = (unsigned char)(v24 >> 0) & 0xFF;
     G_monero_vstate.io_offset += 3;
 }
 
 void monero_io_insert_u16(unsigned int v16) {
     monero_io_hole(2);
-    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 0] = v16 >> 8;
-    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 1] = v16 >> 0;
+    G_monero_vstate.io_buffer[G_monero_vstate.io_offset + 0] = (unsigned char)(v16 >> 8) & 0xFF;
```
Comment (xchapron-ledger) [inline src/monero_io.c:164]: could be done with U2BE_ENCODE from sdk

File: src/monero_ram.c [hunk 7]
```diff
@@ -24,14 +24,14 @@
...
 /* --- "NANO-X" and "NANO-S flow" config --- */
 
 #include "ux.h"
-ux_state_t G_ux;
-bolos_ux_params_t G_ux_params;
+extern ux_state_t G_ux;
+extern bolos_ux_params_t G_ux_params;
 
 #else
 /* --- "NANO-S legacy" config --- */
```
Comment (xchapron-ledger) [inline src/monero_ram.c:37]: To drop probably

File: src/monero_types.h [hunk 8]
```diff
@@ -21,18 +21,11 @@
...
-#define PIN_VERIFIED (!0)
-#elif CX_APILEVEL >= 9
-#define PIN_VERIFIED BOLOS_UX_OK
-#else
-#error CX_APILEVEL not  supported
-#endif
-
-/* cannot send more that F0 bytes in CCID, why? do not know for now
- *  So set up length to F0 minus 2 bytes for SW
+/* The maximal APDU length is 0xF0 bytes, including the 2 bytes needed for the SW response,
```
Comment (xchapron-ledger) [inline src/monero_types.h:24]:
```suggestion
/* The maximal APDU response length is 0xF0 bytes, including the 2 bytes needed for the SW response,
```
and do you now why? Or we are not talking about the response? all this is confusing.

File: src/monero_ux_nano.c [hunk 9]
```diff
@@ -113,11 +90,7 @@ UX_STEP_CB(ux_menu_info_1_step, nn, ui_menu_info_action(0),
 UX_FLOW(ux_flow_info, &ux_menu_info_1_step);
 
 unsigned int ui_menu_info_action(unsigned int value __attribute__((unused))) {
```
Comment (xchapron-ledger) [inline src/monero_ux_nano.c:92]: Drop the function and directly call `ui_menu_main_display()` when needed (nano and stax code).

File: src/monero_ux_nano.c [hunk 10]
```diff
@@ -673,10 +646,14 @@ void ui_menu_any_pubaddr_display(unsigned int value __attribute__((unused)),
             break;
     }
 
-    monero_base58_public_key(G_monero_vstate.ux_address, pub_view, pub_spend, is_subbadress,
-                             paymanetID);
+    int error = monero_base58_public_key(G_monero_vstate.ux_address, pub_view, pub_spend,
+                                         is_subbadress, paymanetID);
+    if (error) {
+        return error;
+    }
```
Comment (xchapron-ledger) [inline src/monero_ux_nano.c:653]: return value is not handle by function that call this.
Maybe monero_base58 could be done before?

File: src/monero_ux_nano.c [hunk 11]
```diff
@@ -686,6 +663,11 @@ void ui_menu_pubaddr_display(unsigned int value) {
     ui_menu_any_pubaddr_display(value, G_monero_vstate.A, G_monero_vstate.B, 0, NULL);
 }
 
+static void exit(void) {
+    memset(&G_monero_vstate, 0, sizeof(G_monero_vstate));
```
Comment (xchapron-ledger) [inline src/monero_ux_nano.c:667]: Not sure this is needed.

File: src/monero_ux_stax_menu.c [hunk 12]
```diff
@@ -86,7 +85,8 @@ static void update_account(void) {
 }
 
 static void exit(void) {
-    os_sched_exit(-1);
+    memset(&G_monero_vstate, 0, sizeof(G_monero_vstate));
```
Comment (xchapron-ledger) [inline src/monero_ux_stax_menu.c:88]: not sure if needed.

File: Makefile [hunk 13]
```diff
@@ -15,188 +15,90 @@
...
-DEFINES       += HAVE_BAGL_FONT_OPEN_SANS_LIGHT_16PX
-endif
-endif
-
-ifeq ($(TARGET_UI),FLOW)
-DEFINES		  += HAVE_UX_FLOW
-endif
-
 # Enabling debug PRINTF
-DEBUG = 0
```
[the remark this answers is not in the document]
Comment (sgliner-ledger) [reply Makefile:119]: Well it prevented passing `DEBUG` as an argument when compiling with `make`. 
But I will remove the line :)

File: src/monero_dispatch.c (hunk 2 above)
Comment (sgliner-ledger) [reply src/monero_dispatch.c:109]: I had a look at the official wallet : Monero GUI which doesnt use it, and also had a look at Feather wallet and Monerujo that don't use it either :)

File: src/monero_api.h [hunk 14]
```diff
@@ -24,7 +24,6 @@
 void __attribute__((noreturn)) send_error_and_kill_app(int sw);
 int monero_apdu_reset(void);
 int monero_apdu_lock(void);
```
Comment (xchapron-ledger) [inline src/monero_api.h:26]: unused too

File: src/monero_ux_nano.c [hunk 15]
```diff
@@ -79,21 +79,14 @@ void settings_show_25_words(void) {
     ui_menu_words_display(0);
 }
 /* -------------------------------- INFO UX --------------------------------- */
-unsigned int ui_menu_info_action(unsigned int value);
-
-UX_STEP_CB(ux_menu_info_1_step, nn, ui_menu_info_action(0),
+UX_STEP_CB(ux_menu_info_1_step, nn, (void)ui_menu_main_display,
```
Comment (xchapron-ledger) [inline src/monero_ux_nano.c:82]: shouldn't it be this instead? (Not sure at all, but looking at was there before, and what's the implementation of `UX_STEP_CB`)
```suggestion
UX_STEP_CB(ux_menu_info_1_step, nn, ui_menu_main_display(),
```

File: src/monero_ux_nano.c [hunk 16]
```diff
@@ -673,17 +639,28 @@ void ui_menu_any_pubaddr_display(unsigned int value __attribute__((unused)),
...
     G_monero_vstate.disp_addr_M = 0;
     G_monero_vstate.disp_addr_m = 0;
-    ui_menu_any_pubaddr_display(value, G_monero_vstate.A, G_monero_vstate.B, 0, NULL);
+    int error = ui_menu_any_pubaddr_display(value, G_monero_vstate.A, G_monero_vstate.B, 0, NULL);
+    if (error) {
+        app_exit();
+    }
+}
+
+static void exit(void) {
```
Comment (xchapron-ledger) [inline src/monero_ux_nano.c:662]: Is this wrapper needed?

File: src/monero_ux_stax_menu.c [hunk 17]
```diff
@@ -86,7 +85,7 @@ static void update_account(void) {
 }
 
 static void exit(void) {
```
Comment (xchapron-ledger) [inline src/monero_ux_stax_menu.c:87]: Needed wrapper?

File: src/monero_ux_nano.c (hunk 15 above)
Comment (sgliner-ledger) [reply src/monero_ux_nano.c:82]: Good catch indeed :+1:

**Review by xchapron-ledger — APPROVED** [review]

**Pull request merged**